dojo/product_type/views.py

```diff
@@ -84,12 +89,51 @@ def add_product_type(request):
     })
 
 
+@user_must_be_authorized(Product_Type, 'view', 'ptid')
+def view_product_type(request, ptid):
+    pt = get_object_or_404(Product_Type, pk=ptid)
+    products = Product.objects.filter(prod_type=pt)
+    add_breadcrumb(title="View Product Type", top_level=False, request=request)
+    return render(request, 'dojo/view_product_type.html', {
+        'name': 'View Product Type',
+        'user': request.user,
+        'pt': pt,
+        'products': products})
+
+
+@user_passes_test(lambda u: u.is_staff)
+def delete_product_type(request, ptid):
+    product_type = get_object_or_404(Product_Type, pk=ptid)
+    form = Delete_Product_TypeForm(instance=product_type)
+
+    if request.method == 'POST':
+        if 'id' in request.POST and str(product_type.id) == request.POST['id']:
+            form = Delete_Product_TypeForm(request.POST, instance=product_type)
+            if form.is_valid():
+                product_type.delete()
+                messages.add_message(request,
+                                     messages.SUCCESS,
+                                     'Product Type and relationships removed.',
+                                     extra_tags='alert-success')
+                return HttpResponseRedirect(reverse('product_type'))
+
+    collector = NestedObjects(using=DEFAULT_DB_ALIAS)
+    collector.collect([product_type])
+    rels = collector.nested()
+
+    add_breadcrumb(title="Delete Product Type", top_level=False, request=request)
```

Comment:
should there be a notification like there currently it in dev? not sure what's happening for other deletion in dojo.

Reply:
Not sure what you mean with "notifications currently in dev". Its exactly the same mechanism as for products: After you clicked "Delete" you get a dialogue with all references. When you click the "Delete" button there the Product Type with all references is gone.

Reply:
You removed the create_notification statement in this PR from delete product type.

Reply:
Oh, these notifications. I will have a look.

Reply:
Yes, the delete notification has to come back, thanks for seeing this @valentijnscholten 

Then I would like to have a "Product Type added" notification as well. Easier for this would be using the event type "Other", more in line with the rest would be to implement a new event type in the notification settings, so a user can decide if he/she wants to notified when a product type is added.

Reply:
@valentijnscholten I decided to go the longer way and implemented a new notification event for added product types. If that's too much I can reverse this.